code/game/objects/machinery/robotic_cradle.dm

```diff
@@ -0,0 +1,311 @@
+
+	if(occupant)
+		to_chat(dragger, span_notice("[src] is already occupied!"))
+		return
+
+	if(machine_stat & (NOPOWER|BROKEN))
+		to_chat(dragger, span_notice("[src] is non-functional!"))
+		return
+
+	if(dragger.skills.getRating(SKILL_ENGINEER) < SKILL_ENGINEER_ENGI && !event)
+		dropped.visible_message(span_notice("[dropped] fumbles around figuring out how to get into \the [src]."),
+		span_notice("You fumble around figuring out how to get into \the [src]."))
+		var/fumbling_time = max(0 , SKILL_TASK_TOUGH - ( SKILL_TASK_EASY * dragger.skills.getRating(SKILL_ENGINEER) ))// 8 secs non-trained, 5 amateur
+		if(!do_after(dropped, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+			return
+
+	dropped.visible_message(span_notice("[dropped] starts climbing into \the [src]."),
+	span_notice("You start climbing into \the [src]."))
+	if(do_after(dropped, 1 SECONDS, FALSE, src, BUSY_ICON_GENERIC))
+		if(occupant)
+			to_chat(dragger, span_notice("[src] is already occupied!"))
+			return
+		dropped.stop_pulling()
+		dropped.forceMove(src)
+		occupant = dropped
+		icon_state = "pod_0"
+		var/implants = list(/obj/item/implant/neurostim)
+		var/mob/living/carbon/human/H = occupant
+		var/doc_dat
+		med_scan(H, doc_dat, implants, TRUE)
+		start_processing()
+		for(var/obj/O in src)
+			qdel(O)
+
+	say("Automatic mode engaged, initialising procedure.")
+	addtimer(CALLBACK(src, PROC_REF(auto_start)), 20 SECONDS)
+
+///Callback to start auto mode on someone entering
+/obj/machinery/robotic_cradle/proc/auto_start()
+	if(repairing)
+		return
+	if(!occupant)
+		say("Occupant missing, procedures canceled.")
+	say("Beginning repair procedure.")
+	repair_op()
+
+/obj/machinery/robotic_cradle/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(!ishuman(user))
+		return // no
+
+	else if(istype(I, /obj/item/healthanalyzer) && occupant) //Allows us to use the analyzer on the occupant without taking him out.
+		var/obj/item/healthanalyzer/J = I
+		J.attack(occupant, user)
+		return
+
+	else if(!istype(I, /obj/item/grab))
+		return
+
+	if(machine_stat & (NOPOWER|BROKEN))
+		to_chat(user, span_notice("[src] is non-functional!"))
+		return
+
+	else if(occupant)
+		to_chat(user, span_notice("[src] is already occupied!"))
+		return
+
+	if(!istype(I, /obj/item/grab))
+		return
+
+	var/obj/item/grab/G = I
+
+	var/mob/M
+	if(ismob(G.grabbed_thing))
+		M = G.grabbed_thing
+	else if(istype(G.grabbed_thing, /obj/structure/closet/bodybag/cryobag))
+		var/obj/structure/closet/bodybag/cryobag/C = G.grabbed_thing
+		if(!C.bodybag_occupant)
+			to_chat(user, span_warning("The stasis bag is empty!"))
+			return
+		M = C.bodybag_occupant
+		C.open()
+		user.start_pulling(M)
+
+	if(!M)
+		return
+
+	else if(!ishuman(M)) // No monkee or beano
+		to_chat(user, span_notice("[src] is compatible with humanoid anatomies only!"))
+		return
+
+	else if(M.abiotic())
+		to_chat(user, span_warning("Subject cannot have abiotic items on."))
+		return
+
+	else if(M.get_species() == "Human")
```

Comment:
Unnecesarry elses here. Also the species check should be changed to `ishumanbasic(M)`.